src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -755,9 +755,12 @@ private void BuildGenericControlInitializerBody(IndentedStringBuilder writer, Xa
 			// OnInitializeCompleted() seems to be used by some older code as a substitute for the constructor for UserControls, which are optimized out of the visual tree.
 			RegisterPartial("void OnInitializeCompleted()");
 
-			writer.AppendLineIndented("NameScope.SetNameScope(this, __nameScope);");
-			writer.AppendLineIndented("var __that = this;");
 			var topLevelControlType = GetType(topLevelControl.Type);
+			if (!IsWindow(topLevelControlType)) // Window is not a DependencyObject
```

Comment:
Should this be `if (IsDependencyObject(topLevelControlType))`?

Reply:
if I am not mistaken, dependency objects have the namescope out of the box - I think I added this because it was complaining Window does not have it